src/loan.rs

```diff
@@ -60,21 +60,177 @@ static COVENANT_PK: &str = "03b9b6059008e3576aad58e05a3a3e37133b05f68cda8535ec09
 /// `covenant_pk`, which is only used to verify that the transaction
 /// data on the witness stack matches the transaction which triggered
 /// the call.
-fn collateral_descriptor(
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+struct CollateralContract {
+    descriptor: Descriptor<PublicKey>,
     borrower_pk: PublicKey,
     lender_pk: PublicKey,
-    timelock: u64,
-    repayment_output: TxOut,
-) -> Result<Descriptor<PublicKey>> {
-    let repayment_output = serialize(&repayment_output).to_hex();
-    let desc = Descriptor::<elements::bitcoin::PublicKey>::from_str(
-        &(format!(
-            "elcovwsh({},or_i(and_v(v:pk({}),after({})),and_v(v:pk({}),outputs_pref({}))))",
-            COVENANT_PK, lender_pk, timelock, borrower_pk, repayment_output,
-        )),
-    )?;
-
-    Ok(desc)
+    timelock: u32, // TODO: Model timelocks as u32
```

Comment:
Is this TODO still relevant? Looks like it is a u32 now?

Reply:
That one disappears with https://github.com/comit-network/baru/pull/25/commits/85795521f639eb0654a22d5f1b730773675a58b6.